### DISCUSSION TITLE
Why is not setting alpha to zero not recommended to skip rendering?

### DISCUSSION THREAD

**Discussion by a-c-sreedhar-reddy** [pixijs/pixijs#7722] [Q&A]
Hi. In the docs I saw the following statement. 
```
Generally, setting alpha to zero is not recommended for purely skipping rendering.
```
Why is not recommended ?

**Comment by gawlas** [comment]
I guess because properties `renderable` and `visible` are mainly responsible for skipping rendering. And not rendering element when worldAlpha is zero is only side effect/optimization.

**Comment by eXponenta** [comment]
`alpha = 0` not prevent all other phase like interactive check and transform updating.

if you want optimise render phase -  use alpha = 0 is not good idea. 

Check difference:

https://www.pixiplayground.com/#/edit/WhXfSZLBUMGBK_JuOUihW

**Comment by a-c-sreedhar-reddy** [comment]
I guess WebGL draw calls would be fired for a sprite with 0 alpha, but would not be fired when renderable is false.?

**Reply by eXponenta** [reply]
Will not be runned for both, but visible = false disable deep tree updating and hit-testing also, renderable = false and alpha = 0 only prevent render pass
